cli/command/container/opts.go

```diff
@@ -899,3 +921,12 @@ func validateAttach(val string) (string, error) {
 	}
 	return val, errors.Errorf("valid streams are STDIN, STDOUT and STDERR")
 }
+
+func convertToNetworkmode(netmode opts.NetworkOpt) container.NetworkMode {
+	networkIDOrName := "default"
+	for _, net := range netmode.Value() {
+		networkIDOrName = net.Target
+		break
```

Comment:
> docker run does not support multiple networks . if user does give multiple networks only the first network is considered.

The reason for only supporting a single network, was that there was no way to pass options per-network (i.e. `--driver-opt` and `--network-alias` would be applied to all networks that you attached the container to)  - also see https://github.com/moby/moby/issues/31964, and issues linked from that.

If the API supports multiple networks on _create_, perhaps we should change that (can be a follow-up to this PR), as this should now be possible from a UX perspective;

```bash
docker run \
  --network name=docknet,alias=web1.0,driver-opt=field1=value1 \
  --network name=second-net,alias=foobar,driver-opt=hello=world \
  nginx:alpine
```